File: code/cpp/Code_5_10_3.cpp
#include <iostream>
using namespace std;

long long N, X, Y;

int main() {
	// 輸入
	cin >> N >> X >> Y;

	// 4 個整數(a, b, c, d) 的全搜尋→輸出答案
	for (int a = 1; a <= N; a++) {
		for (int b = a; b <= N; b++) {
			for (int c = b; c <= N; c++) {
				for (int d = c; d <= N; d++) {
					if (a + b + c + d == X && 1LL * a * b * c * d == Y) {
						cout << "Yes" << endl;
						return 0; // 使程式結束執行
					}
				}
			}
		}
	}
	cout << "No" << endl; // 如果連 1 組都沒發現的話為No
	return 0;
}

File: code/python/Code_5_10_3.py
# 輸入
N, X, Y = map(int, input().split())

# 4 個整數(a, b, c, d) 的全搜尋
flag = False
for a in range(1, N + 1):
	for b in range(a, N + 1):
		for c in range(b, N + 1):
			for d in range(c, N + 1):
				if a + b + c + d == X and a * b * c * d == Y:
					flag = True  # 如果找到答案的話 flag 為 true 

# 輸出答案
if flag == True:
	print("Yes")
else:
	print("No")

# 如 p.256 注 5.10.2 所述，由於 Python 的執行速度較慢，最大情況下執行可能需要數十秒。
# 然而，如果使用相同的程式在 PyPy3 上執行，即使在最大情況下也能在大約 1 秒內得到答案。
# 有關更優化的計算量程式，請參見 Code_5_10_3_extra.py。
